packages/react-router/src/route.ts

```diff
@@ -178,16 +203,16 @@ export type BaseRouteOptions<
     TFullSearchSchema,
     TParams,
     TAllParams,
-    TRouteContextReturn,
-    TParentAllContext,
-    TAllContext,
     TLoaderDeps,
-    TLoaderDataReturn
+    TLoaderDataReturn,
+    TParentAllContext,
+    TRouteContextFn,
+    TBeforeLoadFn
   > & {
     getParentRoute: () => TParentRoute
   }
 
-export interface BeforeLoadContext<
+export interface ContextOptions<
```

Comment:
Just to keep a note here, we should probably type [this beforeLoad](https://github.com/TanStack/router/blob/08b85242f875d619ed285f0ff3dd06db00b30807/packages/react-router/src/router.ts#L1961-L1972) parameter object with a type like we do so for [`the loader`](https://github.com/TanStack/router/blob/08b85242f875d619ed285f0ff3dd06db00b30807/packages/react-router/src/router.ts#L2032-L2054).